architecture/kubernetes_infrastructure.adoc

```diff
@@ -9,43 +9,66 @@
 
 toc::[]
 
-Kubernetes is a system for managing containerized applications across a set of containers or hosts, providing deployment, maintenance, and application-scaling mechanisms. Docker is used to package, instantiate, and run containerized applications. 
+== Overview
+http://kubernetes.io[Kubernetes] is used to manage containerized applications across a set of containers or hosts and provide deployment, maintenance, and application-scaling mechanisms. Docker is used to package, instantiate, and run containerized applications.
 
-A Kubernetes cluster consists of a master and a set of nodes. 
+A Kubernetes cluster consists of a master and a set of nodes.
 
 == Master
+The master is the host or hosts that contain the master components, including the API server, controller manager server, and [sysitem]#etcd#. The master manages link:#node[nodes] in its Kubernetes cluster and schedules link:kubernetes_model.html#pod[pods] to run on nodes.
 
-The Kubernetes master is the host or hosts that contain the master components (API Server, Controller Manger Server, etcd, etc.).
+[cols="1,4"]
+.Master Components
+|===
+|Component |Description
 
-=== API Server
+|API Server
+|The Kubernetes API server validates and configures the data for pods, services, and replication controllers. It also assigns pods to nodes and synchronizes pod information with service configuration.
 
-The Kubernetes API server serves the main API, and validates and configures the data for pods, services, and replication controllers. The Kuberbetes API server also assigns pods to nodes and synchronizes pod information with service configuration. 
+|[sysitem]#etcd#
+|[sysitem]#etcd# stores the persistent master state while other components watch [sysitem]#etcd# for changes to bring themselves into the desired state.
 
-=== etcd
+|Controller Manager Server
+|The controller manager server watches [sysitem]#etcd# for changes to replication controller objects and then uses the API to enforce the desired state.
+|===
```

Comment:
@CowboysFan How do you feel about the master components now being in a table instead of under 3 tiny headings?